### ISSUE TITLE
pygi hangs upon exit

### ISSUE THREAD

**Issue by thazy** [goki/gipy#2]
After starting pygi widgets all three ways (pygi -> import widgets; pygi -i widgets.py; ./widgets.py) everything works, but then upon exiting (two ways: click red dot; pygi -> Quit). Below is example entry in error log after Force Quit: 

FIRST PART OF ERROR LOG:

Date/Time:       2019-08-27 18:02:12 -0600
End time:        2019-08-27 18:02:14 -0600
OS Version:      Mac OS X 10.14.6 (Build 18G87)
Architecture:    x86_64
Report Version:  28

Data Source:     Stackshots
Shared Cache:    0x2f57000 F4C2C6E6-D112-3F8D-A9CA-3C924B088435
Shared Cache:    0 92D6D287-09D6-34A7-9BCC-05865608049B

Command:         pygi
Path:            /usr/local/bin/pygi
Version:         ??? (???)
Parent:          bash [27305] [unique pid 126899]
Responsible:     Terminal [21327] [unique pid 120921]
PID:             28385

Event:           hang
Duration:        1.59s (process was unresponsive for 148 seconds before sampling)
Steps:           16 (100ms sampling interval)

Hardware model:  MacBookPro10,1
Active cpus:     8

Time Awake Since Boot: 440000s

Fan speed:       5929 rpm

--------------------------------------------------
Timeline format: stacks are sorted chronologically
Use -i and -heavy to re-report with count sorting
--------------------------------------------------

Heaviest stack for the main thread of the target process:
  15  runtime.park_m + 174 (pygi + 219518) [0x403597e]
  13  runtime.schedule + 729 (pygi + 219129) [0x40357f9]
  10  runtime.stoplockedm + 138 (pygi + 213194) [0x40340ca]
  10  runtime.notesleep + 227 (pygi + 53283) [0x400d023]
  9  runtime.semasleep + 133 (pygi + 178325) [0x402b895]
  9  runtime.pthread_cond_wait + 81 (pygi + 314417) [0x404cc31]
  9  runtime.asmcgocall + 173 (pygi + 371901) [0x405acbd]
  9  runtime.pthread_cond_wait_trampoline + 16 (pygi + 381488) [0x405d230]
  7  __psynch_cvwait + 10 (libsystem_kernel.dylib + 18538) [0x7fff5ac2086a]
 *5  psynch_cvcontinue + 0 (pthread + 19324) [0xffffff7f8310ab7c]

Process:         pygi [28385] [unique pid 127979]
UUID:            EA5827B9-9F14-3AB2-9558-44795F053878
Path:            /usr/local/bin/pygi
Architecture:    x86_64
Parent:          bash [27305] [unique pid 126899]
Responsible:     Terminal [21327] [unique pid 120921]
UID:             501
Footprint:       199.43 MB
Start time:      2019-08-27 18:02:12 -0600
End time:        2019-08-27 18:02:14 -0600
Num samples:     16 (1-16)
CPU Time:        2.825s (8.4G cycles, 4.8G instructions, 1.75c/i)
Note:            Unresponsive for 148 seconds before sampling
Note:            1 idle work queue thread omitted

  Thread 0x1be5c3           DispatchQueue 1           16 samples (1-16)         priority 46 (base 46)     cpu time 0.901s (2.7G cycles, 1392.5M instructions, 1.91c/i)
  <thread QoS user interactive (requested user interactive), process unclamped, process received importance donation from WindowServer [21117] [unique pid 120711], IO tier 0>
  1   runtime.main + 519 (pygi + 193799) [0x402f507] 1
    1   main.main + 50 (pygi + 12141010) [0x4b941d2] 1
      1   github.com/goki/gi/gimain.Main + 87 (pygi + 10685719) [0x4a30d17] 1
        1   github.com/goki/gi/oswin/driver.Main + 43 (pygi + 10431563) [0x49f2c4b] 1
          1   github.com/goki/gi/oswin/driver.driverMain + 43 (pygi + 10431627) [0x49f2c8b] 1
            1   github.com/goki/gi/oswin/driver/glos.Main + 151 (pygi + 10363511) [0x49e2277] 1
              1   github.com/goki/gi/oswin/driver/glos.(*appImpl).mainLoop + 391 (pygi + 10364807) [0x49e2787] 1
                1   time.Sleep + 291 (pygi + 315027) [0x404ce93] 1
                  1   runtime.(*timersBucket).addtimerLocked + 304 (pygi + 316096) [0x404d2c0] (running) 1
  15  runtime.park_m + 174 (pygi + 219518) [0x403597e] 2-16
    1   runtime.schedule + 729 (pygi + 219129) [0x40357f9] 2
      1   runtime.stoplockedm + 138 (pygi + 213194) [0x40340ca] 2
        1   runtime.notesleep + 227 (pygi + 53283) [0x400d023] 2
          1   runtime.semasleep + 133 (pygi + 178325) [0x402b895] 2
            1   runtime.pthread_cond_wait + 81 (pygi + 314417) [0x404cc31] 2
              1   runtime.asmcgocall + 173 (pygi + 371901) [0x405acbd] 2
                1   runtime.pthread_cond_wait_trampoline + 16 (pygi + 381488) [0x405d230] 2
                  1   __psynch_cvwait + 10 (libsystem_kernel.dylib + 18538) [0x7fff5ac2086a] 2
                   *1   psynch_cvcontinue + 0 (pthread + 19324) [0xffffff7f8310ab7c] (runnable) 2
    1   runtime.schedule + 734 (pygi + 219134) [0x40357fe] 3
     *1   ??? (kernel + 1726816) [0xffffff80003a5960] (runnable) 3
    10  runtime.schedule + 729 (pygi + 219129) [0x40357f9] 4-13
      3   runtime.stoplockedm + 138 (pygi + 213194) [0x40340ca] 4-6
        3   runtime.notesleep + 227 (pygi + 53283) [0x400d023] 4-6
          3   runtime.semasleep + 133 (pygi + 178325) [0x402b895] 4-6
            3   runtime.pthread_cond_wait + 81 (pygi + 314417) [0x404cc31] 4-6
              3   runtime.asmcgocall + 173 (pygi + 371901) [0x405acbd] 4-6
                3   runtime.pthread_cond_wait_trampoline + 16 (pygi + 381488) [0x405d230] 4-6
                  3   __psynch_cvwait + 10 (libsystem_kernel.dylib + 18538) [0x7fff5ac2086a] 4-6
                   *1   call_continuation + 35 (kernel + 1417411) [0xffffff800035a0c3] 4
                     *1   ml_set_interrupts_enabled + 25 (kernel + 2964393) [0xffffff80004d3ba9] (running) 4
                   *2   psynch_cvcontinue + 0 (pthread + 19324) [0xffffff7f8310ab7c] 5-6

**Comment by rcoreilly** [comment]
Yep I've seen that -- I think you need to quit using ctrl+D or something like that.
